### PR TITLE
Merge from upstream through 2020-05-09

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Merge facebook/react-native's main branch into react-native-macos's main branch through 2020-05-09 23:59:59 PDT (specifically, a1ac2518a364ebcd3cc024a22229cadc1791e1c4).

These changes were made by repeatedly running `git merge <commit-id>` and resolving merge conflicts as needed.

One of the more unusual modifications we had to do to ensure compatibility is the addition of the `#if 0` block in React/CoreModules/RCTDevLoadingView.mm. Thankfully, we're blocking out unused code, so it shouldn't be a big deal. This will get cleaned up once we fix this upstream and get microsoft/react-native-macos caught up to that future commit.

## Validation

* yarn lint
* * ✖ 67 problems (0 errors, 67 warnings)
* yarn flow-check-ios
* yarn flow-check-macos
* yarn flow-check-android
* yarn test
* * Test Suites: 128 passed, 128 total
* * Tests: 1 skipped, 1247 passed, 1248 total
* * Snapshots: 573 passed, 573 total
* RNTester compiles on macOS and iOS, and no new regressions have been found

### CODE REVIEW

**Comment by NickGerleman** [comment]
I'm super excited to see macos integrating new upstream changes. Getting macOS up to date will have an incredibly large impact across the RN ecosystem.

Something I've had success with for Windows merges was to limit a single integration to about a week of upstream changes. At that granularity you can inspect individual commits, either looking for changes that intersect with forked code, or doing a narrow search for what change could have caused a regression.

**Comment by HeyImChris** [comment]
"One of the more unusual modifications we had to do to ensure compatibility is the addition of the #if 0 block in React/CoreModules/RCTDevLoadingView.mm. Thankfully, we're blocking out unused code, so it shouldn't be a big deal. This will get cleaned up once we fix this upstream and get microsoft/react-native-macos caught up to that future commit."

Can you elaborate on this? How do we know it's unused? Did fb add a #if 0 and we just don't have that commit synced yet?

**Comment by amgleitman** [comment]
> Something I've had success with for Windows merges was to limit a single integration to about a week of upstream changes. At that granularity you can inspect individual commits, either looking for changes that intersect with forked code, or doing a narrow search for what change could have caused a regression.

I've been doing this locally when moving forward. This PR is just several of these steps forward at a time.

**Comment by amgleitman** [comment]
> "One of the more unusual modifications we had to do to ensure compatibility is the addition of the #if 0 block in React/CoreModules/RCTDevLoadingView.mm. Thankfully, we're blocking out unused code, so it shouldn't be a big deal. This will get cleaned up once we fix this upstream and get microsoft/react-native-macos caught up to that future commit."
> 
> Can you elaborate on this? How do we know it's unused? Did fb add a #if 0 and we just don't have that commit synced yet?

I know `-[RCTDevLoadingView dimColor:]` is unused because the commit that removed that logic was f0dfd35108dd3f092d46b65e77560c35477bf6ba, but this didn't clean up `dimColor:` itself. Also, searching the code for `dimColor` turns up no other results besides this method declaration.

**Review by HeyImChris — APPROVED** [review]

File: Libraries/ART/ARTGroup.m [hunk 1]
```diff
@@ -11,13 +11,19 @@ @implementation ARTGroup
...
 - (void)renderLayerTo:(CGContextRef)context
 {
-
   if (!CGRectIsEmpty(self.clipping)) {
     CGContextClipToRect(context, self.clipping);
   }
 
-  for (ARTNode *node in self.subviews) {
-    [node renderTo:context];
+  for (RCTPlatformView *subview in self.subviews) { // TODO(macOS GH#774)
```
[the remark this answers is not in the document]
Comment (amgleitman) [reply Libraries/ART/ARTGroup.m:18]: The only thing I changed is replacing `UIView` with `RCTPlatformView`.

**Pull request merged**